tests/MongoDB.Bson.TestHelpers/JsonDrivenTests/JsonDrivenTest.cs

```diff
@@ -0,0 +1,134 @@
+﻿/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MongoDB.Bson.TestHelpers.JsonDrivenTests
+{
+    public abstract class JsonDrivenTest
+    {
+        // protected fields
+        protected Exception _actualException;
+        protected BsonDocument _expectedException;
+        protected BsonValue _expectedResult;
+        protected Dictionary<string, object> _objectMap;
```

Comment:
At this level this is a general purpose map that maps "names" to "values".

In the JSON driven transaction tests this objectMap will contain "session0" and "session1".

In future test suites this map might contain other name/value mappings.